core/provider/service_test.go

```diff
@@ -312,6 +312,43 @@ func (s *ServiceTestSuite) TestFetchResources() {
 
 		s.Nil(actualError)
 	})
+
+	s.Run("should upsert filter resources on success", func() {
+		providersWithResourceFilter := []*domain.Provider{
+			{
+				ID:   "1",
+				Type: mockProviderType,
+				URN:  mockProvider,
+				Config: &domain.ProviderConfig{Resources: []*domain.ResourceConfig{
+					{Type: "dataset", Filter: "$resource.urn == 'resource2'"},
```

Comment:
Can you add one more test case with `$resource.name endsWith transaction`  as the filter criteria